src/main/java/org/reservationapplication/model/CoworkingSpace.java

```diff
@@ -1,45 +1,52 @@
-package org.reservationapplication;
+package org.reservationapplication.model;
 
-import java.util.ArrayList;
-import java.util.List;
+
+import org.reservationapplication.repository.CoworkingSpaceRepository;
+import org.reservationapplication.service.CoworkingSpaceServiceImpl;
+
+import static org.reservationapplication.repository.CoworkingSpaceRepository.getNextID;
 
 public class CoworkingSpace {
-    private static long nextId = 0L;
-    private final long coworkingSpaceID;
+    private long coworkingSpaceID;
     private CoworkingSpaceType type;
     private double price;
     private AvailabilityStatus availabilityStatus;
 
-    public CoworkingSpace(CoworkingSpaceService coworkingSpaceService) {
-        this.coworkingSpaceID = nextId++;
-        coworkingSpaceService.addCoworkingSpace(this);
+    public CoworkingSpace() {
+        this.coworkingSpaceID = getNextID();//gets from CoworkingSpaceRepository
```

Comment:
please, just set it manually after creating. now your entity depends on repo, and should be vise versa. How can I make you remove everything regarding id setting from entities?